codec/proto_codec.go

```diff
@@ -160,6 +160,70 @@ func (pc *ProtoCodec) MustUnmarshalJSON(bz []byte, ptr proto.Message) {
 	}
 }
 
+// MarshalInterface is a convenience function for proto marshalling interfaces. It
+// packs the provided value, which must implement proto.Message,
+// in an Any and then marshals it to bytes.
+// NOTE: if you use a concret type, then you should use MarshalBinaryBare instead
+// TODO: check if we can use proto.Message here
```

Comment:
Yes, I think we should!

I believe there are some interfaces that are not Proto.Message yet.
- if it's too big a refactor, we can keep `interface{}` for now
- if not, we can add Proto.Message to all those interfaces.

It'll bit a bit weird that Amino's MarshalInterface also accepts `proto.Message`, but I guess that's fine.

Reply:
It's fine. I've already checked it. I had to updated few interfaces but, but no other code had to be updated since everything which goes into this functions comes from protobuf generated code.